### PR TITLE
Preserve all of /etc/presto during upgrade

### PR DESCRIPTION
**Pull request by ebd2**

### CODE REVIEW

File: prestoadmin/configure_cmds.py [hunk 1]
```diff
@@ -76,33 +75,17 @@ def deploy(rolename=None):
...
-        ensure_parent_directories_exist(local_config_file)
-        if not allow_overwrite and os.path.exists(local_config_file):
-            abort("Refusing to overwrite %s. Use 'overwrite' parameter "
-                  "to overwrite." % (local_config_file))
-        configuration_fetch(file_name, local_config_file)
+def gather_directory():
+    with hide('stdout'):
+        result = sudo(
+            "tar -c -C %s . | base64" % (constants.REMOTE_CONF_DIR,))
+        # Fabric...
```
Comment (cawallin) [inline prestoadmin/configure_cmds.py]: Presumably Fabric adds \r unnecessarily, but please complete that thought :)

File: prestoadmin/configure_cmds.py [hunk 2]
```diff
@@ -76,33 +75,17 @@ def deploy(rolename=None):
...
-def fetch_all(target_directory, allow_overwrite=False):
-    host_config_dir = os.path.join(target_directory, env.host)
-    for file_name in ALL_CONFIG:
-        local_config_file = os.path.join(host_config_dir, file_name)
-        ensure_parent_directories_exist(local_config_file)
-        if not allow_overwrite and os.path.exists(local_config_file):
-            abort("Refusing to overwrite %s. Use 'overwrite' parameter "
-                  "to overwrite." % (local_config_file))
-        configuration_fetch(file_name, local_config_file)
+def gather_directory():
```
Comment (cawallin) [inline prestoadmin/configure_cmds.py]: gather_config_directory()?

File: prestoadmin/configure_cmds.py [hunk 3]
```diff
@@ -76,33 +75,17 @@ def deploy(rolename=None):
...
+    with hide('stdout'):
+        result = sudo(
+            "tar -c -C %s . | base64" % (constants.REMOTE_CONF_DIR,))
+        # Fabric...
+        return result.replace('\r', '')
+
+
+def deploy_all(encoded_tar_conf):
```
Comment (cawallin) [inline prestoadmin/configure_cmds.py]: deploy_config_directory?

File: tests/product/test_server_upgrade.py [hunk 4]
```diff
@@ -132,12 +132,33 @@ def test_upgrade_works_with_symlink(self):
...
         self.run_prestoadmin('configuration deploy')
+        big_files = {}
         for container in self.cluster.all_hosts():
             self.real_installer.assert_installed(self, container)
             self.assert_has_default_config(container)
             self.assert_has_default_catalog(container)
 
+            big_file = self.cluster.exec_cmd_on_host(
+                container,
+                "find /usr -size +2M -ls | "
```
Comment (cawallin) [inline tests/product/test_server_upgrade.py]: Are you sure there'll be a file that matches?

Comment (ebd2) [inline tests/product/test_server_upgrade.py]: There are a fair number of files that match. This runs on a cluster that has presto installed, so there almost 2 dozen copies of guava installed at the very least :-b

Comment (cawallin) [inline tests/product/test_server_upgrade.py]: ok, good then :)

**Pull request merged**